tests/integration/integration-test.sh

```diff
@@ -1,238 +1,304 @@
+  }
+  echo -e "${SUCCESS}"
+}
+
+update_values() { # [EXPRESSION...]
+  for update in "$@"; do
+    yq e -i "${update}" helm/values.yaml
+  done
+}
+
+debug_vaules() {
+  echo "::group::values.yaml"
+  cat helm/values.yaml
+  echo "::endgroup::"
+}
+
+### RUN REGULAR INTEGRATION TEST ####################################
+regular_int_test() {
+  multi_test "regular"
+
+  ### EDGE CASE TAG IN RELEASES AND TARGETS ####################################
+  echo -n "[edge1] Testing edge case of tag defined in both targets and release json file..."
+  DEPLOYED_SHA=$(kubectl get pod pod-rs -o yaml | yq e '.spec.containers[0].image' - | sed 's/.*sha256://')
+  if [[ "${DEPLOYED_SHA}" != 'c5327b291d702719a26c6cf8cc93f72e7902df46547106a9930feda2c002a4a7' ]]; then
+    echo -e "${FAILED}"
+  else
+    echo -e "${SUCCESS}"
+  fi
+
+  ### ALERTING TEST ####################################
+  echo -n "Checking whether alert endpoints have been called successfully..."
+  ENDPOINT_HITS="$(curl -s ${ALERTING_ENDPOINT_IP}:56243 --header 'Content-Type: application/json')"
+  NUMBER_OF_DEPLOYMENTS=$((${DEPLOYMENT_RES["VALID"]} + ${DEPLOYMENT_RES["INVALID"]}))
+  EXPECTED_ENDPOINT_HITS=$(jq -n \
+    --argjson REQUESTS_TO_SLACK_ENDPOINT ${NUMBER_OF_DEPLOYMENTS} \
+    --argjson REQUESTS_TO_OPSGENIE_ENDPOINT ${DEPLOYMENT_RES["VALID"]} \
+    --argjson REQUESTS_TO_KEYBASE_ENDPOINT ${DEPLOYMENT_RES["INVALID"]} \
+    '{
+  "successful_requests_to_slack_endpoint":$REQUESTS_TO_SLACK_ENDPOINT,
+  "successful_requests_to_opsgenie_endpoint": $REQUESTS_TO_OPSGENIE_ENDPOINT,
+  "successful_requests_to_keybase_endpoint": $REQUESTS_TO_KEYBASE_ENDPOINT
+  }')
+  diff <(echo "$ENDPOINT_HITS" | jq -S .) <(echo "$EXPECTED_ENDPOINT_HITS" | jq -S .) >diff.log 2>&1 || true
+  if [[ -s diff.log ]]; then
+    echo -e "${FAILED}"
+    echo "::group::Alerting endpoint diff:"
+    cat diff.log
+    echo "::endgroup::"
+    EXIT="1"
+  else
+    echo -e "${SUCCESS}"
+  fi
+}
+
+### COSIGN TEST ####################################
+cosign_int_test() {
+  multi_test "cosign"
+}
+
+### NAMESPACE VALIDATION TEST ####################################
+namespace_val_int_test() {
+  echo -n "Creating namespaces..."
+  kubectl create namespace ignoredns >/dev/null
+  kubectl label ns ignoredns securesystemsengineering.connaisseur/webhook=ignore >/dev/null
+  kubectl create namespace validatedns >/dev/null
+  kubectl label ns validatedns securesystemsengineering.connaisseur/webhook=validate >/dev/null
+  echo -e "${SUCCESS}"
+
+  multi_test "ignore-namespace-val"
+  update_values '.namespacedValidation.mode="validate"'
+  upgrade_make
+  multi_test "validate-namespace-val"
+}
+
+### DEPLOYMENT TEST ####################################
+deployment_int_test() {
+  multi_test "deployment"
+}
+
+### PRECONFIG TEST ####################################
+pre_config_int_test() {
+  multi_test "pre-config"
+}
+
+case $1 in
+"regular")
+  install_make
+  regular_int_test
+  uninstall_make
+  ;;
+"cosign")
```

Comment:
For running locally, all tests that run successfully, should also uninstall resources they created, in particular Conny. You can hide that behind a flag, as e.g. in previous stress-test

Reply:
But also spawned pods/deployments...

Reply:
```
# Cleanup can be skipped in GitHub Actions, since uninstall is already tested in other integration test
if [[ "${GITHUB_ACTIONS-}" == "true" ]]; then
  exit 0
fi
```